kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutlinePanel.js

```diff
@@ -0,0 +1,260 @@
+/**
+ * Widget for Narrative Outline Panel
+ * @author David Lyon  <dlyon@lbl.gov>
+ * @public
+ */
+define(['jquery', 'base/js/namespace', 'kbwidget', 'kbaseNarrativeControlPanel'], (
+    $,
+    Jupyter,
+    KBWidget,
+    ControlPanel
+) => {
+    'use strict';
+    return new KBWidget({
+        name: 'kbaseNarrativeOutlinePanel',
+        parent: ControlPanel,
+        version: '1.0.0',
+        options: {
+            // If true, all markdown H1-H3 will be outline items
+            // If false, only the first header will be an outline item
+            showAllHeaders: false,
+            // Header height for visibility & scroll functionality
+            headerHeight: 80,
+            // Refresh throttle rate in ms
+            refreshRate: 100,
+        },
+
+        isActive: false,
+        body: undefined,
+        // string representing the items drawn used to eliminate excess dom manipulation
+        drawnState: '',
+
+        init: function (options) {
+            this._super(options);
+            this.body = $(this.body());
+
+            const refresh = this.refresh.bind(this);
+            $('#notebook-container').scroll(refresh);
+            // Trigger refresh on the following jupyter events
+            const refreshJupyterEvents = [
+                'notebook_loaded.Notebook',
+                'notebook_saved.Notebook',
+                'notebook_renamed.Notebook',
+                'command_mode.Notebook', // Triggers on clicking out of editing cell
+                'create.Cell',
+                'delete.Cell',
+                'select.Cell' // Triggers on cell move
+            ];
+            $([Jupyter.events]).on(refreshJupyterEvents.join(' '), refresh);
+
+            // Hide panel header
+            this.$elem.find('.kb-title').hide();
+
+            return this;
+        },
+
+        activate: function () {
+            // Render when the panel is activated
+            this.renderOutline();
+            this.isActive = true;
+        },
+
+        deactivate: function () {
+            this.isActive = false;
+        },
+
+        refresh: function () {
+            if (this.refresh_timeout || !this.isActive) {
+                return;
+            }
+            this.refresh_timeout = setTimeout(() => {
```

Comment:
Kinda picky, but why snake case here? Aside from Jupyter code (which uses snake case throughout both the Python and Javascript stack...), we try to stick to camelCase.

Reply:
🐍

Reply:
Will fix